doc/changelog.md

```diff
@@ -39,6 +39,9 @@
 * Fix the issue of triggering the C++ compiler driver twice.
   [(#594)](https://github.com/PennyLaneAI/catalyst/pull/594)
 
+* Adds lowering pass for `shape` dialect. Allowing for a greater set of programs to be compiled.
+  [(#592)](https://github.com/PennyLaneAI/catalyst/pull/592)
```

Comment:
What kind of programs (what do they look like, what functions are now available, etc...)?

Reply:
I didn't have time to investigate much but mostly those that will lower from StableHLO to the shape dialect. Happy to leave this PR until more time has passed to investigate.

Reply:
It must have come up with some frontend example, or was this need discovered independently?

Reply:
The test is the example.

Reply:
I also added the ticket that lead us to explore this source.

Reply:
Hey @erick-xanadu, I just had a look at the test; any indication to what part of the test circuit was resulting in the bug?

E.g., was it this part?

```python
for i in range(n_qubits - 1):
    if i % 2 == 0:
        qml.CNOT(wires=[i, i + 1])
```

No need to know the details of how it caused a bug, just want to know what a user would need to do to be affected by the bug.

This will help me better communicate in the changelog to users

Reply:
(or this part? `[1::2]`)

Reply:
I think it has to do with the normalization part.

```
    %10 = stablehlo.abs %arg19 : tensor<8xf64>
    %11 = stablehlo.multiply %10, %10 : tensor<8xf64>
    %12 = stablehlo.reduce(%11 init: %9) applies stablehlo.add across dimensions = [0] : (tensor<8xf64>, tensor<f64>) -> tensor<f64>
    %13 = stablehlo.sqrt %12 : tensor<f64>
    %14 = stablehlo.reshape %13 : (tensor<f64>) -> tensor<1xf64>
    %15 = stablehlo.reshape %14 : (tensor<1xf64>) -> tensor<f64>
```

Lowers to HLO's shape operations. These reshape operations come from the AmplitudeEmbedding normalization procedure:

```python
        # normalize
        norm = qml.math.sum(qml.math.abs(features) ** 2, axis=-1)

        if qml.math.is_abstract(norm):
            if normalize or pad_with:
                features = features / qml.math.reshape(qml.math.sqrt(norm), (*shape[:-1], 1))
```

Reply:
Thanks, that's exactly what I was looking for!
So `jnp.reshape` is now supported, as well as at a higher level `AmplitudeEmbedding` with `pad_with` and `normalize` options (which were untested in our existing template test).

Reply:
https://github.com/PennyLaneAI/catalyst/pull/592/commits/ea7fb7eb60636c2b468473830cb586b9270324ef

Reply:
Thanks Erick!